sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala

```diff
@@ -385,7 +385,8 @@ private[spark] object HiveUtils extends Logging {
         hadoopConf = hadoopConf,
         config = configurations,
         barrierPrefixes = hiveMetastoreBarrierPrefixes,
-        sharedPrefixes = hiveMetastoreSharedPrefixes)
+        sharedPrefixes = hiveMetastoreSharedPrefixes,
+        isolationOn = !isCliSessionState)
```

Comment:
One problem,  when cli mode, isolationOn will be false.
SessionState in HiveClientImpl will use CliSessionState based on you buildin hive  version . 
When you use maven to contain hive jars of other version. Can it still work right?